haverhill_311_function/modules/qalert.py

```diff
@@ -15,8 +16,16 @@ def pull():
     data = []
     try:
         url = URL + "?key=" + API_KEY
+        create_date_min = None
+        latest_request = db.QAlertAuditDB().get_latest_request()
```

Comment:
unless u modified the implementation of QAlertAuditDB, in order to establish a session, need to enter the context manager
```suggestion
        with db.QAlertAuditDB() as audit_db:
            latest_request = db.get_latest_request()
```